tests/components/rfxtrx/test_light.py

```diff
@@ -147,82 +156,70 @@ async def test_several_lights(hass, rfxtrx):
     )
     await hass.async_block_till_done()
 
-    assert 3 == len(rfxtrx_core.RFX_DEVICES)
-    device_num = 0
-    for id in rfxtrx_core.RFX_DEVICES:
-        entity = rfxtrx_core.RFX_DEVICES[id]
-        assert entity.signal_repetitions == 3
-        if entity.name == "Living":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Living: off>" == entity.__str__()
-        elif entity.name == "Bath":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Bath: off>" == entity.__str__()
-        elif entity.name == "Test":
-            device_num = device_num + 1
-            assert "off" == entity.state
-            assert "<Entity Test: off>" == entity.__str__()
-
-    assert 3 == device_num
+    assert len(hass.states.async_all()) == 3
 
+    state = hass.states.get("light.test")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Test"
 
-async def test_discover_light(hass, rfxtrx):
-    """Test with discovery of lights."""
+    state = hass.states.get("light.bath")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Bath"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)